### PR TITLE
update git extract test structure

### PR DESCRIPTION
**Pull request by charlierudolph**
@kevgo

Took both ideas from #315 

* move the bigger features into folders, moved the smaller features into the same file 
* update feature names and links to user value

### CODE REVIEW

**Comment by charlierudolph** [comment]
Tried it out and we can't have two `Feature` lines in one feature file. Gherkin fails to parse it.
Joining the smaller tests into one file and just converting the two features into two scenarios within a feature (largely for those that are just catching errors)

File: features/git-extract/branch_exists_locally.feature [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
+    Then it runs the Git commands
+      | BRANCH  | COMMAND           |
+      | feature | git fetch --prune |
+    And I get the error "A branch named 'existing-feature' already exists"
+    And I am still on the "feature" branch
+    And I still have an uncommitted file with name: "uncommitted" and content: "stuff"
 
 
-  Scenario: result
+  Scenario: without open changes
```
Comment (kevgo) [inline features/git-extract/branch_exists_locally.feature]: Oh, I love this!

File: features/git-extract/branch_exists_remotely.feature [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
+    And I get the error "A branch named 'existing-feature' already exists"
+    And I am still on the "feature" branch
+    And I still have an uncommitted file with name: "uncommitted" and content: "stuff"
 
 
-  Scenario: result
+  Scenario: without open changes
+    Given I have a feature branch named "feature"
+    And my coworker has a feature branch named "existing-feature"
+    And I am on the "feature" branch
```
Comment (kevgo) [inline features/git-extract/branch_exists_remotely.feature]: You could extract the first 3 lines into a background block here.

File: features/git-extract/no_branch_name.feature [hunk 3]
```diff
@@ -15,10 +15,21 @@ Feature: git extract: abort if no branch name is given (with open changes)
...
 
-  Scenario: result
+  Scenario: without open changes
+    Given I have a feature branch named "feature"
+    And the following commits exist in my repository
+      | BRANCH  | LOCATION | MESSAGE            | FILE NAME        |
+      | main    | remote   | remote main commit | remote_main_file |
+      | feature | local    | feature commit     | feature_file     |
+      |         |          | refactor commit    | refactor_file    |
+    And I am on the "feature" branch
```
Comment (kevgo) [inline features/git-extract/no_branch_name.feature]: Everything until this line can go into `Background`

**Comment by kevgo** [comment]
This is awesome @charlierudolph! Please coordinate who works on what if you want help with cleaning up the other feature specs.

**Comment by kevgo** [comment]
:ship:

File: features/git-extract/no_branch_name.feature (hunk 3 above)
Comment (charlierudolph) [inline features/git-extract/no_branch_name.feature]: done

File: features/git-extract/branch_exists_remotely.feature (hunk 2 above)
Comment (charlierudolph) [inline features/git-extract/branch_exists_remotely.feature]: done

**Comment by charlierudolph** [comment]
This was easy. I'm fine doing it the rest of the way. I'll keep the PR's limited to a single command.

**Pull request closed without merging**